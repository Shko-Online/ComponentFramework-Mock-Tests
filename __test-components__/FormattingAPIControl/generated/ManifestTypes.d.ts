/*
    Copyright (c) 2022 Betim Beja and Shko Online LLC
    Licensed under the MIT license.
*/

// Define IInputs and IOutputs Type. They should match with ControlManifest.
export interface IInputs {
    currencyInput: ComponentFramework.PropertyTypes.NumberProperty;
    dateInput: ComponentFramework.PropertyTypes.DateTimeProperty;
    decimalInput: ComponentFramework.PropertyTypes.DecimalNumberProperty;
    integerInput: ComponentFramework.PropertyTypes.WholeNumberProperty;
}
export interface IOutputs {
    currencyInput?: number;
    dateInput?: Date;
    decimalInput?: number;
    integerInput?: number;
}
