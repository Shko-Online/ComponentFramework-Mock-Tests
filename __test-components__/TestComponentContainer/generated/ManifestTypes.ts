/*
    Copyright (c) 2022 Betim Beja and Shko Online LLC
    Licensed under the MIT license.
*/

// Define IInputs and IOutputs Type. They should match with ControlManifest.
export interface IInputs {
    controlValue: ComponentFramework.PropertyTypes.StringProperty;
}
export interface IOutputs {
    controlValue?: string;
}
