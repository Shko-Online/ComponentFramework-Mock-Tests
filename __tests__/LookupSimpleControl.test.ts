import * as sinon from 'sinon';

import { ComponentFrameworkMockGenerator } from '@shko-online/componentframework-mock/ComponentFramework-Mock-Generator';
import { LookupSimpleControl } from '@powerapps-samples/lookup-simple-control/LookupSimpleControl';
import { IInputs, IOutputs } from '@powerapps-samples/lookup-simple-control/LookupSimpleControl/generated/ManifestTypes';
import { LookupPropertyMock } from '@shko-online/componentframework-mock/ComponentFramework-Mock/PropertyTypes/LookupProperty.mock';

describe("ControlStateAPI tests", () => {
    let mockGenerator: ComponentFrameworkMockGenerator<IInputs, IOutputs>;
    beforeEach(() => {
        const container = document.createElement('div');

        mockGenerator = new ComponentFrameworkMockGenerator(
            LookupSimpleControl,
            {
                controlValue: LookupPropertyMock,
                controlValue1: LookupPropertyMock
            },
            container);
        mockGenerator.context.parameters.controlValue.raw = [{
            entityType: 'mocked_entity',
            id: "00000000-0000-0000-0000-000000000001",
            name: 'Test Betim'
        }];
        mockGenerator.context.parameters.controlValue1.raw = [{
            entityType: 'mocked_entity',
            id: "00000000-0000-0000-0000-000000000002",
            name: 'Test Betim2'
        }];
        document.body.appendChild(container);
    }
    )

    afterEach(() => {
        document.body.innerHTML = null;
    })

    it("Init should work", () => {
        mockGenerator.ExecuteInit();
        sinon.assert.calledOnce(mockGenerator.control.init);
        expect(document.body).toMatchSnapshot();
    })

    it("Update View should Work", () => {

        mockGenerator.ExecuteInit();
        mockGenerator.ExecuteUpdateView();
        sinon.assert.calledOnce(mockGenerator.control.init);
        sinon.assert.calledOnce(mockGenerator.control.updateView);
        expect(document.body).toMatchSnapshot();
    })

    it("Click should Work", () => {       
        mockGenerator.ExecuteInit();
        mockGenerator.ExecuteUpdateView();
        expect(document.body).toMatchSnapshot();
       
        const button = mockGenerator.container.querySelector("button");
        var evt = document.createEvent("Event");
        evt.initEvent("click", false, true);
        button.dispatchEvent(evt);
   
        mockGenerator.ExecuteUpdateView();     
            
        expect(document.body).toMatchSnapshot();
    })
});