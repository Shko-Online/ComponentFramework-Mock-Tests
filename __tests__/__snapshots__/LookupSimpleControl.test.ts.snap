// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlStateAPI tests Click should Work 1`] = `
<body>
  <div>
    <div>
      <label>
        Simple Lookup Control
      </label>
      <br />
      <br />
      <label>
        name: Test Betim entityType: mocked_entity id: 00000000-0000-0000-0000-000000000001
      </label>
      <br />
      <br />
      <button />
      <br />
      <br />
      <label>
        name: Test Betim2 entityType: mocked_entity id: 00000000-0000-0000-0000-000000000002
      </label>
      <br />
      <br />
      <button />
      <br />
      <br />
    </div>
  </div>
</body>
`;

exports[`ControlStateAPI tests Click should Work 2`] = `
<body>
  <div>
    <div>
      <label>
        Simple Lookup Control
      </label>
      <br />
      <br />
      <label>
        name: Test Betim entityType: mocked_entity id: 00000000-0000-0000-0000-000000000001
      </label>
      <br />
      <br />
      <button />
      <br />
      <br />
      <label>
        name: Test Betim2 entityType: mocked_entity id: 00000000-0000-0000-0000-000000000002
      </label>
      <br />
      <br />
      <button />
      <br />
      <br />
    </div>
  </div>
</body>
`;

exports[`ControlStateAPI tests Init should work 1`] = `
<body>
  <div>
    <div>
      <label>
        Simple Lookup Control
      </label>
      <br />
      <br />
      <label />
      <br />
      <br />
      <button />
      <br />
      <br />
      <label />
      <br />
      <br />
      <button />
      <br />
      <br />
    </div>
  </div>
</body>
`;

exports[`ControlStateAPI tests Update View should Work 1`] = `
<body>
  <div>
    <div>
      <label>
        Simple Lookup Control
      </label>
      <br />
      <br />
      <label>
        name: Test Betim entityType: mocked_entity id: 00000000-0000-0000-0000-000000000001
      </label>
      <br />
      <br />
      <button />
      <br />
      <br />
      <label>
        name: Test Betim2 entityType: mocked_entity id: 00000000-0000-0000-0000-000000000002
      </label>
      <br />
      <br />
      <button />
      <br />
      <br />
    </div>
  </div>
</body>
`;
